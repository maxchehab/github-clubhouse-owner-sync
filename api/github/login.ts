import { NextApiResponse, NextApiRequest } from 'next';

import lambda from '../../common/util/lambda.util';

export default lambda(async (_req: NextApiRequest, res: NextApiResponse) => {
  res
    .status(302)
    .setHeader(
      'location',
      `https://github.com/login/oauth/authorize?scope=read:org&client_id=${process.env.GITHUB_OAUTH_CLIENT_ID}`,
    );

  return res.send(null);
});
