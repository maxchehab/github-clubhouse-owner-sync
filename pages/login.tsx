import { Pane, Button, Heading } from 'evergreen-ui';

export default () => (
  <Pane
    alignItems={'center'}
    display={'flex'}
    height={'100vh'}
    justifyContent={'center'}
    width={'100vw'}
    flexDirection={'column'}
  >
    <Heading paddingBottom={40} size={900}>
      GitHub Clubhouse Owner Sync
    </Heading>
    <a href={'/api/github/login'}>
      <Button appearance={'primary'} height={40}>
        Login with GitHub
      </Button>
    </a>
  </Pane>
);
