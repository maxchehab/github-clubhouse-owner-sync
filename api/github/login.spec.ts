import axios from 'axios';
import http from 'http';
import listen from 'test-listen';

import closeServer from '../../test/close-server';
import createTestLambda from '../../test/create-test-lambda';
import login from './login';

describe('api/github/login', () => {
  let server: http.Server;
  let url: string;

  beforeAll(async () => {
    server = createTestLambda(login);
    url = await listen(server);
  });

  afterAll(async () => {
    await closeServer(server);
  });

  describe('with GET api/github/login', () => {
    it('returns 302, redirect to GitHub', async () => {
      const { status, headers } = await axios.get(url, {
        maxRedirects: 0,
        validateStatus: () => true,
      });

      expect(status).toEqual(302);
      expect(headers.location).toMatchSnapshot();
    });
  });
});
