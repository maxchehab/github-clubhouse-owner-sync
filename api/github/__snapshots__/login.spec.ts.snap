// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/github/login with GET api/github/login returns 302, redirect to GitHub 1`] = `"https://github.com/login/oauth/authorize?scope=read:org&client_id=test-github-oauth-client-id"`;
